projects/igniteui-angular/src/lib/core/styles/components/tree/_tree-theme.scss

```diff
@@ -0,0 +1,356 @@
+    $palette: $default-palette,
+    $schema: $light-schema,
+    $background: null,
+    $background-selected: null,
+    $background-active: null,
+    $background-active-selected: null,
+    $background-disabled: null,
+    $foreground: null,
+    $foreground-selected: null,
+    $foreground-active: null,
+    $foreground-active-selected: null,
+    $foreground-disabled: null,
+    $drop-indicator-color: null,
+    $border-color: null,
+) {
+    $name: 'igx-tree';
+    $tree-schema: ();
+
+    @if map-has-key($schema, $name) {
+        $tree-schema: map-get($schema, $name);
+    } @else {
+        $tree-schema: $schema;
+    }
+
+    $theme: apply-palette($tree-schema, $palette);
+
+    @if not($foreground) and $background {
+        $foreground: text-contrast($background);
+    }
+
+    @if not($foreground-selected) and $background-selected {
+        $foreground-selected: text-contrast($background-selected);
+    }
+
+    @if not($foreground-active) and $background-active {
+        $foreground-active: text-contrast($background-active);
+    }
+
+    $_hover-color: --var($theme, '_hover-color');
+
+    @return extend($theme, (
+        name: $name,
+        palette: $default-palette,
+
+        background: $background,
+        background-selected: $background-selected,
+        background-active: $background-active,
+        background-active-selected: $background-active-selected,
+        background-disabled: $background-disabled,
+
+        foreground: $foreground,
+        foreground-selected: $foreground-selected,
+        foreground-active: $foreground-active,
+        foreground-active-selected: $foreground-active-selected,
+        foreground-disabled: $foreground-disabled,
+
+        drop-indicator-color: $drop-indicator-color,
+        border-color: $border-color,
+        _hover-color: $_hover-color,
+    ));
+}
+
+/// @param {Map} $theme - The theme used to style the component.
+/// @requires {mixin} igx-css-vars
+/// @requires --var
+@mixin igx-tree($theme) {
+    @include igx-css-vars($theme);
+
+    $left: if-ltr(left, right);
+    $right: if-ltr(right, left);
+
+    $variant: map-get($theme, variant);
+    $indigo-theme: $variant == 'indigo-design';
+
+    $type: map-get($theme, type);
+    $dark-variant: $type == 'dark';
+
+    $node-indent: (
+        comfortable: rem(24px),
+        cosy: rem(16px),
+        compact: rem(12px)
+    );
+
+    $node-height: (
+        comfortable: rem(50px),
+        cosy: rem(40px),
+        compact: rem(32px)
+    );
+
+    $icon-size: rem(24px);
+    $icon-space: rem(8px);
+
+    $drop-indicator-width: (
+        comfortable: calc(100% - ((#{map-get($node-indent, 'comfortable')} * 2) + (#{$icon-size} + #{$icon-space}))),
+        cosy: calc(100% - ((#{map-get($node-indent, 'cosy')} * 2) + (#{$icon-size} + #{$icon-space}))),
+        compact: calc(100% - ((#{map-get($node-indent, 'compact')} * 2) + (#{$icon-size} + #{$icon-space})))
+    );
+
+    %tree-display {
+        display: block;
```

Comment:
Give this `z-index: 0` to establish a new stacking context specific to the tree.